Snakeyaml should not be included as bundle

Snakeyaml should be embedded as private JAR in the bundle in https://github.com/Netcentric/accesscontroltool/blob/b19bdeb48145ab53514e5b0671c1ebceaf82fc18/accesscontroltool-bundle/pom.xml#L271
instead of being added as public bundle in https://github.com/Netcentric/accesscontroltool/blame/develop/accesscontroltool-package/pom.xml#L110.

The latter leads to conflict with AEMaaCS which ships with another version of the Snakeyaml bundle (1.24).

The issue can not be reproduced with a local AEM SDK as there the affected bundle `com.adobe.granite.toggle.impl.remote:1.0.74` nor snakeyaml is not deployed there OOTB.

This is only a problem since version 2.6.0 which upgraded Snakeyaml from 1.13 to 1.27. AEMaaCS uses version 1.24. As the OSGi installer only does upgrades but not downgrades, the bundle version 1.13 was never effective but version 1.24 was. Therefore this only a problem in reality since 2.6.0 of AC Tool.
Unfortunately snakeyaml 1.27 is not fully backwards compatible to 1.24. Therefore the following issue might be observed in AEMaaCS with AC Tool 2.6.0 and 2.6.1
```
Error during instantiation of the implementation object (org.apache.felix.log.LogException: java.lang.reflect.InvocationTargetException: null)
org.apache.felix.log.LogException: java.lang.reflect.InvocationTargetException: null
.
.

Caused by: java.lang.LinkageError: loader constraint violation: when resolving method "org.yaml.snakeyaml.Yaml.<init>(Lorg/yaml/snakeyaml/constructor/BaseConstructor;Lorg/yaml/snakeyaml/representer/Representer;)V" the class loader org.apache.felix.framework.BundleWiringImpl$BundleClassLoader @1625ff2d (instance of org.apache.felix.framework.BundleWiringImpl$BundleClassLoader, child of org.apache.sling.feature.launcher.spi.Launcher$LauncherClassLoader @5b7ea70d org.apache.sling.feature.launcher.spi.Launcher$LauncherClassLoader) of the current class, com/adobe/granite/toggle/impl/mappers/YamlConverter, and the class loader org.apache.felix.framework.BundleWiringImpl$BundleClassLoader @7bec2432 (instance of org.apache.felix.framework.BundleWiringImpl$BundleClassLoader, child of org.apache.sling.feature.launcher.spi.Launcher$LauncherClassLoader @5b7ea70d org.apache.sling.feature.launcher.spi.Launcher$LauncherClassLoader) for the method's defining class, org/yaml/snakeyaml/Yaml, have different Class objects for the type org/yaml/snakeyaml/constructor/BaseConstructor used in the signature
at com.adobe.granite.toggle.impl.mappers.YamlConverter.<init>(YamlConverter.java:31) [com.adobe.granite.toggle.impl.remote:1.0.74]
```

--- a/accesscontroltool-bundle/pom.xml
+++ b/accesscontroltool-bundle/pom.xml
@@ -268,7 +268,7 @@ Bundle-SymbolicName: biz.netcentric.cq.tools.accesscontroltool.bundle
 # export all versioned packages by default
 -exportcontents: ${packages;VERSIONED}
 # Embed (https://groups.google.com/g/bndtools-users/c/NAdU0_X8OKY)
--includeresource: tomcat-el-api-[0-9.]*.jar;lib:=true,tomcat-jasper-el-[0-9.]*.jar;lib:=true
+-includeresource: tomcat-el-api-[0-9.]*.jar;lib:=true,tomcat-jasper-el-[0-9.]*.jar;lib:=true,snakeyaml-[0-9.]*.jar;lib:=true
 # broad version range for jackrabbit oak versions (as long as we need to support Oak < 1.7.12, see https://github.com/Netcentric/accesscontroltool/issues/435 and https://issues.apache.org/jira/browse/OAK-6945
 Import-Package: org.apache.jackrabbit.oak.spi.security.authentication.external;resolution:=optional;version="[0,3)",org.apache.jackrabbit.oak.spi.security.authentication.external.basic;resolution:=optional;version="[0,2)",org.apache.jackrabbit.oak.*;version="[0,2)",com.adobe.granite.crypto;resolution:=optional,com.adobe.granite.jmx.annotation;resolution:=optional,com.day.cq.security.util;resolution:=optional,* ]]></bnd>
                 </configuration>

--- a/accesscontroltool-bundle/pom.xml
+++ b/accesscontroltool-bundle/pom.xml
@@ -303,6 +303,7 @@ Import-Package: org.apache.jackrabbit.oak.spi.security.authentication.external;r
                         </goals>
                         <phase>verify</phase>
                         <configuration>
+                            <useMavenDependencies>false</useMavenDependencies>
                             <bndruns>
                                 <include>minimum-aem.bndrun</include>
                             </bndruns>
@@ -315,6 +316,7 @@ Import-Package: org.apache.jackrabbit.oak.spi.security.authentication.external;r
                         </goals>
                         <phase>verify</phase>
                         <configuration>
+                            <useMavenDependencies>false</useMavenDependencies>
                             <bndruns>
                                 <include>maximum-aem.bndrun</include>
                             </bndruns>

--- a/accesscontroltool-package/pom.xml
+++ b/accesscontroltool-package/pom.xml
@@ -107,9 +107,6 @@
                             <target>/apps/netcentric/actool/install/27</target>
                         </embedded>
                         <embedded>
-                            <artifactId>snakeyaml</artifactId>
-                        </embedded>
-                        <embedded>
                             <artifactId>bcprov-jdk15on</artifactId>
                         </embedded>
                         <embedded>

--- a/target-osgi-environment/maximum-environment/pom.xml
+++ b/target-osgi-environment/maximum-environment/pom.xml
@@ -253,6 +253,19 @@
             <version>1.2.4</version> <!-- uses Oak 1.3.7 -->
             <scope>provided</scope>
         </dependency>
+        <!-- OSGi bundles shipped with the ACTool package -->
+        <dependency>
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bcpkix-jdk15on</artifactId>
+            <version>${bouncycastle.version}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <artifactId>bcprov-jdk15on</artifactId>
+            <groupId>org.bouncycastle</groupId>
+            <version>${bouncycastle.version}</version>
+            <scope>provided</scope>
+        </dependency>
         <dependency>
             <groupId>com.adobe.aem</groupId>
             <artifactId>uber-jar</artifactId>

--- a/target-osgi-environment/minimum-environment/pom.xml
+++ b/target-osgi-environment/minimum-environment/pom.xml
@@ -234,6 +234,19 @@
             <version>2.1.6</version>
             <scope>provided</scope>
         </dependency>
+        <!-- OSGi bundles shipped with the ACTool package -->
+        <dependency>
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bcpkix-jdk15on</artifactId>
+            <version>${bouncycastle.version}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <artifactId>bcprov-jdk15on</artifactId>
+            <groupId>org.bouncycastle</groupId>
+            <version>${bouncycastle.version}</version>
+            <scope>provided</scope>
+        </dependency>
         <dependency>
             <groupId>com.adobe.aem</groupId>
             <artifactId>uber-jar</artifactId>

